### PR TITLE
WIP: platform, documentation code changes input for input component

### PR DESCRIPTION
**Pull request by Lokanathan-k**
#### Please provide a link to the associated issue.

https://github.com/SAP/fundamental-ngx/issues/2298

#### Please provide a brief summary of this pull request.

The text input is used to enter single of text. When empty, it can hold a placeholder similar to a select component. You can define the height and width of the text input and also determine specific behaviour when handling texts.

The Input has the same features and functionality as the Input component in Fundamental NGX: Core. As such, the Platform Input component will be a wrapper around the Core Input component and will provide identical bindings as Input. Additionally, since this is an input element, it needs to implement the FormFieldControl as described in the FormGroup Layout or extend existing BaseInputComponent.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] update `README.md`
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by Lokanathan-k** [comment]
> id, name, value, contentsize (compact/cozy), state, disabled should be part of base component (ask about disabled to frank). Status will be part of baseInput, as it is showing form valid/invalid status.

on implementing the interface FormFieldControl we are bound to override few function offered from the formfield control. if ew dont implement this iterface in our component and include other common parameters we are facing a circular dependency on the components.

File: libs/platform/src/lib/components/form/input/input.component.ts [hunk 1]
```diff
@@ -39,15 +42,37 @@ export type InputType = 'text' | 'number' | 'email';
...
+    styleUrls: ['input.component.scss'],
     providers: [
-        { provide: FormFieldControl, useExisting: InputComponent, multi: true }
+        { provide: FormFieldControl, useExisting: PlatformInputComponent, multi: true }
     ]
 })
-export class InputComponent extends BaseInput {
+export class PlatformInputComponent extends BaseInput implements CssClassBuilder {
 
     @Input()
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/input/input.component.ts:52]: Missing typedocs

File: libs/platform/src/lib/components/form/input/input.component.ts [hunk 2]
```diff
@@ -39,15 +42,37 @@ export type InputType = 'text' | 'number' | 'email';
...
     type: InputType = 'text';
 
+    @Input()
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/input/input.component.ts:55]: Missing typedocs

File: libs/platform/src/lib/components/form/input/input.component.ts [hunk 3]
```diff
@@ -39,15 +42,37 @@ export type InputType = 'text' | 'number' | 'email';
...
+    readonly: boolean = false;
+
+    /** Whether the input is disabled. */
+    @Input()
+    disabled: boolean = false;
+
+    @Input()
+    compact: boolean = false;
+    
+    @Input()
```
Comment (InnaAtanasova) [inline libs/platform/src/lib/components/form/input/input.component.ts:69]: Missing typedocs

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
+<fd-docs-section-title [id]="'inputdef'" [componentName]="'Input'">
```
Comment (salarenko) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html:1]: Inputs attribute values should use kebab-case, same for `line: 10`.

**Comment by InnaAtanasova** [comment]
There are some empty scss files. Could you remove them? Please rebase master and resolve the conflicts. @Vanessa-Cusmich could you take a look at the documentation part. Thank you

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<fd-docs-section-title [id]="'inputdef'" [componentName]="'input'">
+	Input
+</fd-docs-section-title>
+<description>This example shows Input default implementaion </description>
```
Comment (Vanessa-Cusmich) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html:4]: This doesn't appear on the Preview page. 

Can you also please provide more description for each category of examples, like in the Popover component you reference below?

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+<component-example>
+	<fd-platform-input-example></fd-platform-input-example>
+</component-example>
+<code-example [exampleFiles]="defaultInputType"></code-example>
+
+<fd-docs-section-title [id]="'inputValid'" [componentName]="'inputValidation'">
+	Input States with Messages Components.
```
Comment (Vanessa-Cusmich) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html:11]: Should be 'Input States With Message Components' (all words should be capitalized without punctuation) :)

**Comment by fkolar** [comment]
I am missing a point what is the goal of this PR and I dont really understand those platform changes. we had one input already under Form/ package, how this one is different ?

**Comment by Lokanathan-k** [comment]
> e had one input already under Form/ p

this PR is the documentation part of the input form the form Component

**Comment by fkolar** [comment]
> > e had one input already under Form/ p
> 
> this PR is the documentation part of the input from the form Component

It's not. You are doing some significant changes , renaming classes, extending iterfaces, etc...

File: libs/platform/src/lib/components/form/input/input.component.ts [hunk 7]
```diff
@@ -39,15 +42,42 @@ export type InputType = 'text' | 'number' | 'email';
...
     templateUrl: 'input.component.html',
     changeDetection: ChangeDetectionStrategy.OnPush,
+    styleUrls: ['input.component.scss'],
     providers: [
-        { provide: FormFieldControl, useExisting: InputComponent, multi: true }
+        { provide: FormFieldControl, useExisting: PlatformInputComponent, multi: true }
     ]
 })
-export class InputComponent extends BaseInput {
+export class PlatformInputComponent extends BaseInput implements CssClassBuilder {
```
Comment (fkolar) [inline libs/platform/src/lib/components/form/input/input.component.ts:50]: Please stop this renaming, Checkout other platform components, yout prefix component, we prefix only module. the same for files names

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<fd-docs-section-title [id]="'inputdef'" [componentName]="'input'">
```
Comment (salarenko) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.html:1]: Input attribute values should use kebab-case

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-example.component.html [hunk 9]
```diff
@@ -0,0 +1,27 @@
+<div fd-form-item>
+    <label fd-form-label for="input-52">
```
Comment (salarenko) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-example.component.html:2]: Broken id / for attribute binding

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-validation-example.component.html [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+        <fd-form-message [type]="'information'">
+            Pellentesque metus lacus commodo eget justo ut rutrum varius nunc
+        </fd-form-message>
+    </fd-form-input-message-group>
+</div>
+<div fd-form-item>
+    <label fd-form-label for="input-52">
+        Disabled Input
+    </label>
+    <fdp-input InputType="number" placeholder="Enter value here" [disabled]="true"></fdp-input>
```
Comment (salarenko) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-validation-example.component.html:50]: Why sometimes `fdp-input` uses `InputType` and sometimes `type`?

**Comment by stefanoScalzo** [comment]
Can you make the title more descriptive please

File: libs/platform/src/lib/components/form/input/input.component.ts [hunk 11]
```diff
@@ -34,12 +42,42 @@ export type InputType = 'text' | 'number' | 'email';
...
+
+    /** Whether the input is disabled. */
+    @Input()
+    disabled: boolean = false;
+
+    /** to define the compactness of the componenent. */
+    @Input()
+    compact: boolean = false;
+    
+    /**calass variable to add the class names.  */
```
Comment (stefanoScalzo) [inline libs/platform/src/lib/components/form/input/input.component.ts:72]: typo

File: libs/platform/src/lib/components/form/input/input.component.scss [hunk 12]
```diff
@@ -0,0 +1 @@
+@import "~fundamental-styles/dist/input";
```
Comment (JKMarkowski) [inline libs/platform/src/lib/components/form/input/input.component.scss:1]: Do you need to import it? We have this import in FormControlDirective

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-header/platform-input-header.component.ts [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+@Component({
+  selector: 'fd-platform-input-header',
+  templateUrl: './platform-input-header.component.html'
+})
+export class PlatformInputHeaderComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (JKMarkowski) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-header/platform-input-header.component.ts:12]: Let's remove unused auto generated code

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-validation-example.component.ts [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  selector: 'fd-platform-input-validation-example',
+  templateUrl: './platform-input-validation-example.component.html'
+})
+export class PlatformInputValidationExampleComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
+
```
Comment (JKMarkowski) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-validation-example.component.ts:13]: Let's remove unused auto generated code

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-example.component.ts [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+@Component({
+  selector: 'fd-platform-input-example',
+  templateUrl: './platform-input-example.component.html'
+})
+export class PlatformInputExampleComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (JKMarkowski) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-example/platform-input-example.component.ts:12]: Let's remove unused auto generated code

File: apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.ts [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+            fileName: 'platform-input-validation-example',
+            component: 'PlatformInputValidationExampleComponent'
+        }
+    ];
+
+
+  constructor() { }
+
+  ngOnInit(): void {
+  }
```
Comment (JKMarkowski) [inline apps/docs/src/app/platform/component-docs/platform-input/platform-input-docs.component.ts:45]: Let's remove unused auto generated code

**Comment by fkolar** [comment]
@Lokanathan-k  can you park this PR? And finish first this info label, close it with comment and then you can continue..

**Comment by droshev** [comment]
@Lokanathan-k The PR is getting old; has conflicts and the commits are way too many.

**Pull request closed without merging**